### PR TITLE
[FLINK-13877][hive] Support Hive version 2.1.0 and 2.1.1

### PR DESCRIPTION
**Pull request by xuefuz**
## What is the purpose of the change

Support  Hive 2.1.x versions (2.1.0 and 2.1.1).

## Brief change log

  - Added new shims for each version
  - Adjusted shim inheritance heirarchy
  - Relocated one shimmed method

## Verifying this change

This change is already covered by existing tests, and manually ran the tests for all versions.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV210.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * Shim for Hive version 2.1.0.
+ */
+public class HiveShimV210 extends HiveShimV122 {
+
+	@Override
+	public IMetaStoreClient getHiveMetastoreClient(HiveConf hiveConf) {
+		try {
+			Class<?>[] constructorArgTypes = {HiveConf.class};
+			Object[] constructorArgs = {hiveConf};
+			Method method = RetryingMetaStoreClient.class.getMethod("getProxy", HiveConf.class,
```
Comment (lirui-apache) [inline flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV210.java]: Can we call `public static IMetaStoreClient getProxy(HiveConf hiveConf, boolean allowEmbedded)` instead? Then `HiveShimV230` probably doesn't have to implement its own `getHiveMetastoreClient `?

**Comment by bowenli86** [comment]
can we add test profiles for new Hive versions? 

I’d be more comfortable if we can at least run them manually.

**Comment by xuefuz** [comment]
Added profile. @bowenli86 Can you take another look?

File: flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV210.java (hunk 1 above)
Comment (lirui-apache) [reply flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV210.java]: Thanks for the details. Maybe we can add some comments for that?

**Pull request closed without merging**